ngraph/core/shape_inference/include/rnn_cell_shape_inference.hpp

```diff
@@ -0,0 +1,51 @@
+// Copyright (C) 2018-2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+#pragma once
+#include <openvino/op/rnn_cell.hpp>
+
+namespace ov {
+namespace op {
+namespace util {
+template <class T>
+void validate_input_rank(const RNNCellBase* op, const std::vector<T>& input) {
+    enum { X, initial_hidden_state, W, R, B };
+
+    // Verify static ranks for all inputs
+    for (size_t i = 0; i < input.size(); i++) {
+        NODE_VALIDATION_CHECK(dynamic_cast<const ngraph::Node*>(op),
+                              (input[i].rank().is_static()),
+                              "RNNCellBase supports only static rank for input tensors. Input ",
+                              i);
+    }
+
+    // Verify input dimension against values provided in spec (LSTMCell_1.md)
+    for (size_t i = 0; i < input.size(); i++) {
+        if (i == B) {
+            // verify only B input dimension which is 1D
+            NODE_VALIDATION_CHECK(dynamic_cast<const ngraph::Node*>(op),
+                                  (input[i].rank().get_length() == 1),
+                                  "RNNCellBase B input tensor dimension is not correct.");
+        } else {
+            // Verify all other input dimensions which are 2D tensor types
+            NODE_VALIDATION_CHECK(dynamic_cast<const ngraph::Node*>(op),
+                                  (input[i].rank().get_length() == 2),
```

Comment:
Seems more suitable here, please apply to all the suitable places in this PR
```suggestion
                                  (input[i].size() == 2),
```